### ISSUE TITLE
refactor periodbase, add multiharmonic AoV and fast-chi periodograms

### ISSUE THREAD

**Issue by waqasbhatti** [waqasbhatti/astrobase#10]
We should refactor periodbase so it has a bunch of submodules, one each for each period-search algorithm. Importing periodbase will hoist the main methods up to the main namespace, so no existing code based on periodbase should suffer.

In addition, we should add:

- http://adsabs.harvard.edu/abs/1996ApJ...460L.107S
- http://adsabs.harvard.edu/cgi-bin/bib_query?arXiv:0901.1913

Other stuff to investigate later:

- http://iopscience.iop.org/article/10.1088/2041-8205/811/2/L34
- http://iopscience.iop.org/article/10.1088/0004-637X/812/1/18

**Comment by waqasbhatti** [comment]
The refactoring is done and tested. Doing
```python
from astrobase import periodbase
```
wiill still work fine. To import individual submodules for each period-search method:
```python
from astrobase.periodbase import spdm, saov, kbls, zgls
```

Now working on adding in the other two period-search methods.

**Comment by waqasbhatti** [comment]
- added multiharmonic AoV in 5e92346e452b895b5efef47346277275ed9b81cf, fixed bugs in f17bd8e80a1689de298539adb7fb1b5ed081e561
- probably won't add the fast-chi one; it seems a bit redundant
- also added a McQuillan+ (2013a, 2014) ACF period finder
- may add the Graham+ (2013) conditional entropy period-finder later as it looks useful based on: http://adsabs.harvard.edu/abs/2013MNRAS.434.3423G

Will close this for now.
